src/components/autocomplete/js/autocompleteController.js

```diff
@@ -64,6 +66,18 @@ function MdAutocompleteCtrl ($scope, $element, $mdUtil, $mdConstant, $mdTheming,
     $mdUtil.initOptionalProperties($scope, $attrs, { searchText: null, selectedItem: null });
     $mdTheming($element);
     configureWatchers();
+
+    if ($scope.componentId) {
+      removeFromRegistry = $mdComponentRegistry.register({
+        // Restricted API interface for the component registry.
+        // We don't want to provide the whole controller as API.
+        clearCache: function() {
+          // Run the clearCache method in the controller context.
+          ctrl.clearCache();
```

Comment:
Yes, you're right. I decided to you the in-line implementation. Because a controller function would be unnecessary, when `$mdAutocomplete` is present.